File: rb/features.rb
# Coinpaprika SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module CoinpaprikaFeatures
  def self.make_feature(name)
    case name
    when "base"
      CoinpaprikaBaseFeature.new
    when "test"
      CoinpaprikaTestFeature.new
    else
      CoinpaprikaBaseFeature.new
    end
  end
end

File: py/features.py
# Coinpaprika SDK feature factory

from feature.base_feature import CoinpaprikaBaseFeature
from feature.test_feature import CoinpaprikaTestFeature


def _make_feature(name):
    features = {
        "base": lambda: CoinpaprikaBaseFeature(),
        "test": lambda: CoinpaprikaTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
